File: lib/models/game.dart
import 'turn.dart';
import 'user.dart';

class Game {
  final String gameId;
  DateTime? started;
  DateTime? finished;
  final List<User> players;
  List<Turn> turns = [];

  Game(this.gameId, this.players);
  Game.initAll(this.gameId, this.players, this.turns, this.started, this.finished);
  void addTurn(Turn turn) {
    turns.add(turn);
  }
}

File: src/models/game.ts
import { Turn } from '~/models/turn';
import { GuestUser, User } from '~/models/user';

export interface Game {
    id: string;
    players: (Omit<User, 'friends'> | GuestUser)[];
    turns: Turn[];
    started: Date;
    finished: Date;
    startingScore: number;
}
